adapter-base/src/main/java/org/eclipse/hono/adapter/auth/device/X509AuthProvider.java

```diff
@@ -52,6 +52,9 @@ public X509AuthProvider(final CredentialsClient credentialsClient, final Tracer
      * The JSON object passed in is required to contain a <em>subject-dn</em> and a
      * <em>tenant-id</em> property.
      * <p>
+     * The JSON property <em>auth-id-template</em> is available only if an auth-id template
+     * is configured in the tenant's trust anchor for the given subject DN.
+     * <p>
```

Comment:
IMHO this should be modified to be more instructive to the reader, e.g.
> If the JSON object does not contain an <em>auth-id-template</em> property, the value of the <em>subject-dn</em> property is used as the authentication identifier of the returned credentials object. Otherwise, ...